src/deck/drivers/src/usddeck.c

```diff
@@ -565,60 +622,13 @@ int usddeckFrequency(void)
 
 void usddeckTriggerLogging(void)
 {
-  uint32_t ticks = xTaskGetTickCount();
-  xSemaphoreTake(logBufferMutex, portMAX_DELAY);
-
-  // trigger writing once there is some data
-  if (logBufferReadPtr < logBufferWritePtr && xHandleWriteTask) {
-    vTaskResume(xHandleWriteTask);
-  }
-
-  int dataSize = 2 + 4 + usdLogConfig.numBytes;
-
-  // if we reach the end of the buffer, start from the beginning again
-  if (logBufferWritePtr + dataSize >= logBuffer + usdLogConfig.bufferSize) {
-    logBufferWritePtr = logBuffer;
-  }
-
-  // only write if we have enough space
-  if (logBufferReadPtr < logBufferWritePtr + dataSize) {
-    /* write data into buffer */
-    uint16_t event_id = 0xFFFF;
-    memcpy(logBufferWritePtr, &event_id, sizeof(event_id));
-    logBufferWritePtr += sizeof(event_id);
-    memcpy(logBufferWritePtr, &ticks, 4);
-    logBufferWritePtr += 4;
-    for (int i = 0; i < usdLogConfig.numSlots; ++i) {
-      logVarId_t varid = usdLogConfig.varIds[i];
-      switch (logGetType(varid)) {
-        case LOG_UINT8:
-        case LOG_INT8:
-        {
-          memcpy(logBufferWritePtr, logGetAddress(varid), sizeof(uint8_t));
-          logBufferWritePtr += sizeof(uint8_t);
-          break;
-        }
-        case LOG_UINT16:
-        case LOG_INT16:
-        {
-          memcpy(logBufferWritePtr, logGetAddress(varid), sizeof(uint16_t));
-          logBufferWritePtr += sizeof(uint16_t);
-          break;
-        }
-        case LOG_UINT32:
-        case LOG_INT32:
-        case LOG_FLOAT:
-        {
-          memcpy(logBufferWritePtr, logGetAddress(varid), sizeof(uint32_t));
-          logBufferWritePtr += sizeof(uint32_t);
-          break;
-        }
-        default:
-          ASSERT(false);
-      }
+  // TODO: use a fixed lookup rather than a loop
```

Comment:
So, leaving a TODO like this will quell once conscience. What is the plan of ensuring this TODO: gets fixed? Should an issue be created? What is stopping us from doing it now?

Reply:
Good catch - forgot about it. Fixed now.